modules/extensions/src/mask/mask.js

```diff
@@ -20,17 +20,19 @@ export default class MaskExtension extends LayerExtension {
     };
   }
 
-  draw({uniforms, moduleParameters}) {
+  draw({uniforms, context, moduleParameters}) {
     uniforms.mask_maskByInstance = this.state.maskByInstance;
     const {maskId} = this.props;
     const {maskChannels} = moduleParameters;
+    const {viewport} = context;
     if (maskChannels && maskChannels[maskId]) {
-      const {index, bounds} = maskChannels[maskId];
-
-      const bl = this.projectPosition([bounds[0], bounds[1], 0]);
-      const tr = this.projectPosition([bounds[2], bounds[3], 0]);
+      const {index, bounds, coordinateOrigin, coordinateSystem} = maskChannels[maskId];
       uniforms.mask_enabled = true;
       uniforms.mask_channel = index;
+
+      const opts = {viewport, coordinateOrigin, coordinateSystem};
+      const bl = projectPosition([bounds[0], bounds[1], 0], opts);
+      const tr = projectPosition([bounds[2], bounds[3], 0], opts);
```

Comment:
You need `fromCoordinate*`: https://github.com/visgl/deck.gl/blob/master/modules/core/src/shaderlib/project/project-functions.js#L105

Reply:
A bit confused here. The bounds are specified in the coordinate system of the mask. What should I be passing into the `fromCoordinate*` parameters?

Reply:
`fromCoordinateSystem` (source) is from the mask layer's props.
`coordinateSystem` (target) is from the current layer's props.

Otherwise you are projecting to the source layer's common space, which isn't necessarily the same as the one you are using it in.

Reply:
I've made the change, however I had to pass in a `null` `modelMatrix` to make it work. Does that mean that we should also pass through the `modelMatrix` of the mask layer

Reply:
Let's leave the model matrix as null for now. I think `layer.getBounds()` should apply the matrix of defined.